### PR TITLE
feat(526): Toxic exposure destructive modal pattern

### PR DESCRIPTION
**Pull request by LinnJS**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders

## Summary

This PR enhances the toxic exposure conditions validation in the disability benefits claim form by:

1. Adding required field validation to ensure at least one condition is selected
2. Improving error handling and validation logic for toxic exposure conditions
3. Adding a new custom page component (ToxicExposureChoicePage) to handle destructive actions when users opt out of toxic exposure claims

The changes improve the user experience by:
- Providing clearer validation feedback
- Adding a confirmation modal when users opt out of toxic exposure claims
- Implementing proper data cleanup when users change their selections
- Adding success alerts and proper navigation handling

The solution addresses potential data inconsistency issues and improves the overall form flow by properly handling the destructive action of removing toxic exposure data.

## Related issue(s)

- Related: [#110031](https://github.com/department-of-veterans-affairs/va.gov-team/issues/110031)

## Testing done

Old behavior:
- No validation for required field selection
- No confirmation when opting out of toxic exposure claims
- Potential data inconsistency when changing selections

New behavior:
- Validates that at least one condition is selected
- Shows confirmation modal when opting out with existing data
- Properly cleans up data when changing selections
- Shows success alert after data deletion
- Handles navigation properly in both form and review contexts

Testing completed:
1. Unit tests for validation logic
2. Integration tests for form flow
3. Manual testing of:
   - Initial condition selection
   - Opting out with no existing data
   - Opting out with existing data
   - Navigation in form context
   - Navigation in review context
   - Success alert display and dismissal
   - Modal confirmation flow

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | N/A    | N/A   |
| Desktop | N/A    | N/A   |

## What areas of the site does it impact?

This change impacts the disability benefits claim form, specifically the toxic exposure section. It affects:
- The conditions selection page
- The review page when editing toxic exposure conditions
- The form navigation flow

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature
- [x] No sensitive information is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated
- [ ] Screenshot of the developed feature is added
- [ ] Accessibility testing has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature has a monitor built into Datadog or Grafana

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Known gaps in testing that may raise risk:
1. Edge cases around rapid navigation between pages
2. Performance impact of state management with large form datasets
3. Browser compatibility testing for the modal and alert components
4. Load testing with multiple concurrent users

Would appreciate specific review of:
1. State management approach in ToxicExposureChoicePage
2. Error handling strategy
3. Navigation timing with setTimeout usage
4. Accessibility implementation of the modal and alert components

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js [hunk 1]
```diff
@@ -8,28 +8,69 @@ import {
   validateTEConditions,
 } from '../../content/toxicExposure';
 import { formTitle, makeConditionsUI } from '../../utils';
+import ToxicExposureChoicePage from './toxicExposureChoicePage';
+
+// TODO: Add destructive modal for toxic exposure
```
Comment (LinnJS) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js:13]: Add the modal via custom page, adding the custom page configuration here.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 2]
```diff
@@ -0,0 +1,520 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import FormNavButtons from 'platform/forms-system/src/js/components/FormNavButtons';
```
Comment (LinnJS) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:3]: New custom page, but onChange it not adding conditions to pageData... Something is misaligned probably the data structure somewhere or just not plugged in all the way.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      }
+    },
+    [showDeletedToxicExposureConfirmation],
+  );
+
+  const modalRef = useRef(null);
+  useEffect(
+    () => {
+      if (showDeleteToxicExposureModal && modalRef.current) {
+        const modalHeading = document.querySelector('h4');
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:91]: I assume this is how we did it elsewhere but do we maybe need a more specific query selector if we're querying directly on document? (Since there could be other h4s now or in the future)

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        scrollAndFocus(modalHeading);
+      }
+    },
+    [showDeleteToxicExposureModal],
+  );
+
+  const hasToxicExposureData = () => {
+    return toxicExposureKeys.some(key => {
+      if (key === 'conditions') return false; // Skip conditions as it's the trigger
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:100]: Nit: I might not make this assumption in case this changes for some reason

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js [hunk 5]
```diff
@@ -19,7 +22,13 @@ export const uiSchema = {
       updateUiSchema: makeTEConditionsUISchema,
     }),
   },
-  'ui:validations': [validateTEConditions],
+  // View fields for tracking state changes - hidden from user
+  'view:previousToxicExposureConditions': {
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js:26]: Double checking: these are still necessary because of the review page is that correct? So even with a custom page we have to have some properties in the schema?

**Review by NB28VT — DISMISSED** [review]
Nice! Looks like you have pretty much everything hooked up and well tested. Happy to pair if there's still something not working.

I think most of my notes are either nits or non-issues. I was curious about all the type checking code (which I think JR did as well). I'm not sure how necessary that is but I may be missing something.

It worries me a bit because it's a pretty complex bit of code and should be covered by some type of utility. I worry we could accidentally introduce bugs this way that may be tough to find but that's only looking at it with a glance so take that with a grain of salt

File: src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  'Remove toxic exposure conditions from your claim?';
+
+export const deleteToxicExposureModalDescription =
+  "If you choose to not claim any conditions related to toxic exposure, we'll delete this information from your claim:";
+
+export const deleteToxicExposureModalContent = (
+  <>
+    <ul>
+      <li>
+        <strong>Toxic exposure conditions</strong>
```
[the remark this answers is not in the document]
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx:13]: Yeah I think its because <strong> and <em> have semantic meaning to screen readers and <b> does not. I think they use <strong> over <em> to be more explicit. 🙂

File: src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      </li>
+      <li>
+        <strong>Gulf War service locations and dates (1990 and 2001)</strong>
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx:16]: Is there a reason for not dynamically generating this list?

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+  'specifyOtherExposures',
+];
+
+export const sectionsOfToxicExposure = [
+  'Toxic exposure conditions',
+  'Gulf War service locations and dates (1990 and 2001)',
+  'Agent Orange exposure locations and dates',
+  'Other toxic exposure details and dates',
+];
+
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:53]: This list and the content of `deleteToxicExposureModalContent` in `src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx `appear to be identical. It may make sense for each to have a shared source, in case any future changes are made to content.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+      // Keep only the 'none' condition selected
+      deepClone.toxicExposure.conditions = { none: true };
+    }
+
+    setFormData(deepClone);
+  };
+
+  // If the user opts to deselect "none" and they are asked in the modal to delete their toxic exposure data,
+  // they can close the modal or click the Cancel button. Doing so should revert the "none" checkbox back to unchecked
+  const revertNoneConditionSelection = () => {
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:144]: Was this a design request? I would argue opposed to doing this, but that we only revert a deselection of a condition after a user chooses not to proceed with removing associated toxic exposure data.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+      deepClone.toxicExposure = {};
+    }
+    deepClone.toxicExposure.conditions = toxicExposureSelections;
+
+    setFormData(deepClone);
+  };
+
+  const shouldShowDeleteToxicExposureModal = () => {
+    return (
+      data.toxicExposure?.conditions?.none === true && hasToxicExposureData()
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:161]: Should selecting 'none' be the only trigger? What about deselecting all conditions, but not selecting 'none'?

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 11]
```diff
@@ -27,6 +27,7 @@ import {
   addForm0781,
   addForm0781V2,
   addForm8940,
+  addToxicExposure,
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/submit-transformer.js:30]: Is this still a WIP? `addToxicExposure` is not in `/utils/submit`.

File: src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx [hunk 12]
```diff
@@ -0,0 +1,305 @@
...
+        veteranInformation: { name: 'John Doe' },
+      };
+
+      const result = addToxicExposure(formData);
+
+      expect(result).to.deep.equal(formData);
+    });
+  });
+
+  describe('when user has toxic exposure data but no conditions', () => {
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx:19]: Perhaps modify the description to specify the user has no "toxic exposure" conditions.

File: src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx [hunk 13]
```diff
@@ -0,0 +1,305 @@
...
+        },
+      };
+
+      const result = addToxicExposure(formData);
+
+      expect(result).to.deep.equal(formData);
+    });
+  });
+
+  describe('when user has mixed conditions including "none"', () => {
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx:69]: I don't think that this is possible, given the current UI validations. It is not possible to select a condition AND "none".

**Review by freeheeling — CHANGES_REQUESTED** [review]
Raised some questions for your review.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 14]
```diff
@@ -0,0 +1,358 @@
...
+      // Check if at least one condition is selected
+      const conditions = data.toxicExposure?.conditions || {};
+      const hasSelection = Object.values(conditions).some(
+        value => value === true,
+      );
+
+      if (!hasSelection) {
+        scrollToFirstError({ focusOnAlertRole: true });
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:194]: Good catch for some reason I thought it was required, I will double check.

[the remark this answers is not in the document]
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx]: This was correct, code and test were updated to reflect.

File: src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx (hunk 13 above)
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx:69]: Removed this test case.

File: src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx (hunk 7 above)
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx]: Done ✅

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js (hunk 5 above)
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js:26]: Yeah pretty sure its required by forms system

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:146]: After talking to @rubensun @freeheeling and team in parking lot we are moving to make this consistent with other destruction modals. The main concern is around content and framing of this interaction to users.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 15]
```diff
@@ -0,0 +1,358 @@
...
+    const deepClone = cloneDeep(data);
+
+    if (deepClone.toxicExposure) {
+      toxicExposureKeys.forEach(key => {
+        if (key !== 'conditions') {
+          delete deepClone.toxicExposure[key];
+        }
+      });
+      // Keep only the 'none' condition selected
+      deepClone.toxicExposure.conditions = { none: true };
```
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx]: Put another way, wouldn't setting `none` to **true** only make sense if that option were initially selected, prior to clicking **Continue**?

The way I'm thinking about this is a user, who has originally associated a condition with a toxic exposure event, later deselects that condition on the conditions page and then clicks **Continue**. That action triggers the modal, which, if confirmed, shouldn't change any of the checkbox values on the page.

After confirming date deletion via the modal, the `none` condition should **only** be selected if, before clicking **Continue**, the user had also selected the `none` checkbox. That checkbox corresponds to an actual checkbox on the paper form that should **only** be marked if the user intentionally selected the corresponding `none` option on the online form. Not just because they chose to remove a condition.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 16]
```diff
@@ -0,0 +1,358 @@
...
+    }
+
+    setFormData(deepClone);
+  };
+
+  // If the user opts to deselect "none" and they are asked in the modal to delete their toxic exposure data,
+  // they can close the modal or click the Cancel button. Doing so should revert the "none" checkbox back to unchecked
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx]: This comment is confusing as there is no "reversion" to unchecked if `none` was previously in a deselected state.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:146]: I'm still unclear here as to why, any time the modal is closed, we are setting `none` to **false**, regardless of what the value was prior to rendering the modal.

File: src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+  'Gulf War service locations and dates (1990 and 2001)',
+  'Agent Orange exposure locations and dates',
+  'Other toxic exposure details and dates',
+];
+
+export const deleteToxicExposureModalTitle =
+  'Remove condition related to toxic exposure?';
+
+export const deleteToxicExposureModalDescription =
+  "If you choose to remove flat feet as a condition related to toxic exposure, we'll delete information about:";
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx:13]: Did you mean to include "flat feet"?

Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/content/toxicExposureChoiceContent.jsx:13]: Use curly aposrophe.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 18]
```diff
@@ -0,0 +1,366 @@
...
+
+export const toxicExposureChoicePageTitle =
+  'Option to claim conditions related to toxic exposure';
+
+/**
+ * List of all toxic exposure related keys that need to be deleted
+ * when user opts out of toxic exposure claims.
+ * @constant {string[]}
+ */
+export const toxicExposureKeys = [
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:34]: Seems like this could be a shared `const` with that included in src/applications/disability-benefits/all-claims/utils/submit.js.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 19]
```diff
@@ -0,0 +1,366 @@
...
+  deleteToxicExposureModalDescription,
+  deleteToxicExposureModalTitle,
+} from '../../content/toxicExposureChoiceContent';
+import { formTitle, showToxicExposureDestructionModal } from '../../utils';
+
+export const toxicExposureChoicePageTitle =
+  'Option to claim conditions related to toxic exposure';
+
+/**
+ * List of all toxic exposure related keys that need to be deleted
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:30]: Does 'conditions' get deleted? Should it be included in this array?

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 20]
```diff
@@ -0,0 +1,366 @@
...
+    },
+    [showDeleteToxicExposureModal],
+  );
+
+  const hasToxicExposureData = () => {
+    return toxicExposureKeys.some(key => {
+      if (key === 'conditions') return false; // Skip conditions as it's the trigger
+      if (!(key in (data?.toxicExposure || {}))) return false;
+
+      const value = data.toxicExposure[key];
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:99]: I think all that would need to be evaluated is whether `value` is a non-empty Object.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx [hunk 21]
```diff
@@ -0,0 +1,366 @@
...
+      });
+      // If user explicitly selected "none", keep it selected
+      // Otherwise, clear all conditions
+      if (data.toxicExposure?.conditions?.none === true) {
+        deepClone.toxicExposure.conditions = { none: true };
+      } else {
+        deepClone.toxicExposure.conditions = {};
+      }
+    }
+
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx:139]: I assume it has been verified, after deleting data, that it is possible to re-add data.

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js [hunk 22]
```diff
@@ -20,8 +22,13 @@ export const uiSchema = {
       updateUiSchema: makeTEConditionsUISchema,
     }),
   },
-  'ui:validations': [validateTEConditions],
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureConditions.js:23]: Is this validation implemented elsewhere now?

File: src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx [hunk 23]
```diff
@@ -0,0 +1,285 @@
...
+      };
+
+      const result = addToxicExposure(formData);
+
+      expect(result).to.deep.equal(formData);
+    });
+  });
+
+  describe('when user has toxic exposure data but no toxic exposure conditions', () => {
+    it('should return the form data unchanged', () => {
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/utils/addToxicExposure.unit.spec.jsx:20]: This shouldn't be possible. Or, if a condition was selected that enabled a location to be selected, and then the condition was deselected, confirming removal of the condition via the destructive action modal should also remove all other `toxicExposure` keys.

File: src/applications/disability-benefits/all-claims/tests/utils/utils.unit.spec.jsx
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/utils/utils.unit.spec.jsx:1]: What are the reasons for the changes not related to `showToxicExposureDestructionModal` in this file?

File: src/applications/disability-benefits/all-claims/utils/submit.js [hunk 24]
```diff
@@ -752,6 +752,52 @@ export const addFileAttachments = formData => {
   return { ...clonedData, ...(attachments.length && { attachments }) };
 };
 
+export const addToxicExposure = formData => {
+  // If no toxic exposure data exists, return unchanged
+  if (!formData.toxicExposure) {
+    return formData;
+  }
+
+  // If user selected only "none" for conditions, remove all toxic exposure data except conditions
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/utils/submit.js:761]: Is this an additional failsafe if the destructive action modal does not remove toxic exposure keys?

File: src/applications/disability-benefits/all-claims/tests/components/ToxicExposureChoicePage.unit.spec.jsx [hunk 25]
```diff
@@ -0,0 +1,867 @@
...
+
+      await waitFor(() => {
+        const modal = document.querySelector('va-modal');
+        expect(modal).to.exist;
+        expect(modal.getAttribute('visible')).to.equal('true');
+        expect(goForward.called).to.be.false;
+      });
+    });
+  });
+
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/components/ToxicExposureChoicePage.unit.spec.jsx:246]: What happens in the case where a condition and "none" are selected, whether or not there are existing toxic exposure data?

File: src/applications/disability-benefits/all-claims/tests/components/ToxicExposureChoicePage.unit.spec.jsx [hunk 26]
```diff
@@ -0,0 +1,867 @@
...
+        // Check modal content lists
+        expect(modal.innerHTML).to.contain(
+          'Gulf War service locations and dates (1990 and 2001)',
+        );
+        expect(modal.innerHTML).to.contain(
+          'Agent Orange exposure locations and dates',
+        );
+        expect(modal.innerHTML).to.contain(
+          'Other toxic exposure details and dates',
+        );
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/components/ToxicExposureChoicePage.unit.spec.jsx:753]: So, these other toxic exposure locations and details are listed, even though only 1990 Gulf War corresponds to the data?

File: src/applications/disability-benefits/all-claims/utils/submit.js (hunk 24 above)
Comment (LinnJS) [reply src/applications/disability-benefits/all-claims/utils/submit.js]: Move this logic into another PR
